test/unit/extension/lib/shopgate/CartExtensionPipelineTest.js

```diff
@@ -76,4 +76,8 @@ describe('CartExtensionPipeline - unit', () => {
 
     return subjectUnderTest.getCheckoutUrl().should.eventually.equal(expectedUrl)
   })
+  it('update products should return true', function () {
+    bigCommerceCartRepositoryMock.expects('updateItems').once().returns()
+    return subjectUnderTest.updateProducts([{productId: '1', quantity: 1}]).should.eventually.equal(false)
```

Comment:
What you need to do here also is to test that the error was logged, which means you'll need to have a mock on pipeline context and you'll need to be able to call the anonymous function.
You can do this by using callsFake, like this:
```
bigCommerceCartRepositoryMock.expects('updateItems').once().callsFake((items, notify) => {
      notify({
        reason: 'fake reason',
        item: {
          itemId: '21321',
          quantity: 1
        }
      })
    })
```

Reply:
Added a test for error condition.